File: VB/ReportingBackend/Controllers/WebDocumentViewerController.vb
Imports System.Linq
Imports System.Web.Mvc
Imports System.Web.Routing
Imports DevExpress.Web.Mvc.Controllers
Imports ReportingBackend.Services

Namespace ReportingBackend.Controllers
    Public Class WebDocumentViewerController
        Inherits WebDocumentViewerApiController

        #Region "WebDocumentViewer Overrides"
        ' This action processes each request from the Web Document Viewer.
        Public Overrides Function Invoke() As ActionResult
            Response.AddHeader("Access-Control-Allow-Origin", "*") ' Allow cross-domain requests.
            Return MyBase.Invoke()
        End Function
        #End Region

        ' Create an action to provide the Web Document Viewer with available reports.
        Public Function GetReportList() As ActionResult
            Dim reports = (New CustomReportResolver()).GetAvailableReports().ToArray()
            Response.AddHeader("Access-Control-Allow-Origin", "*") ' Allow cross-domain requests.
            Return New JsonResult() With {.ContentType = "application/json", .Data = reports}
        End Function
    End Class
End Namespace

File: CS/ReportingBackend/Controllers/WebDocumentViewerController.cs
using System.Linq;
using System.Web.Mvc;
using System.Web.Routing;
using DevExpress.Web.Mvc.Controllers;
using ReportingBackend.Services;

namespace ReportingBackend.Controllers {
    public class WebDocumentViewerController : WebDocumentViewerApiController {
        #region WebDocumentViewer Overrides
        // This action processes each request from the Web Document Viewer.
        public override ActionResult Invoke() {
            Response.AddHeader("Access-Control-Allow-Origin", "*");// Allow cross-domain requests.
            return base.Invoke();
        }
        #endregion

        // Create an action to provide the Web Document Viewer with available reports.
        public ActionResult GetReportList() {
            var reports = new CustomReportResolver().GetAvailableReports().ToArray();
            Response.AddHeader("Access-Control-Allow-Origin", "*");// Allow cross-domain requests.
            return new JsonResult() { ContentType = "application/json", Data = reports };
        }
    }
}